tests/pytests/test_replicate.py

```diff
@@ -211,3 +211,85 @@ def testDropWith__FORCEKEEPDOCS():
 
     env.assertEqual(master.execute_command('KEYS', '*'), ['doc1'])
     env.assertEqual(slave.execute_command('KEYS', '*'), ['doc1'])
+
+
+def testExpireDocs():
+  env = initEnv()
+  master = env.getConnection()
+  master.execute_command('DEBUG', 'SET-ACTIVE-EXPIRE', '0')
+  slave = env.getSlaveConnection()
+  slave.execute_command('DEBUG', 'SET-ACTIVE-EXPIRE', '0')
+
+  '''
+  This test creates creates an index and two documents and check they
+  exist on both shards.
+  One of the documents is expired.
+  The test checks the document is removed from both master and slave.
+  '''
+
+  for i in range(2):
+    sortby_cmd = [] if i == 0 else ['SORTBY', 't']
+    master.execute_command('FT.CREATE', 'idx', 'SCHEMA', 't', 'TEXT')
+    master.execute_command('HSET', 'doc1', 't', 'bar')
+    master.execute_command('HSET', 'doc2', 't', 'foo')
+    
+    # both docs exist
+    res = master.execute_command('FT.SEARCH', 'idx', '*', *sortby_cmd)
+    env.assertEqual(res, [2, 'doc1', ['t', 'bar'], 'doc2', ['t', 'foo']])
+    checkSlaveSynced(env, slave, ('FT.SEARCH', 'idx', '*', 'NOCONTENT'), [2, 'doc1', 'doc2'], time_out=5)
```

Comment:
Any reason why you run with `NOCONTENT` here?

Reply:
We don't expect a change and IMO it is more readable.
Changed